orttraining/orttraining/training_ops/cuda/nn/layer_norm_impl.cu

```diff
@@ -293,7 +302,7 @@ struct SharedMemory<double> {
 };
 }  // namespace
 
-template <typename T, typename U>
+template <typename T, typename U, bool simplified>
```

Comment:
I think the forward kernel and its helper functions are repeated here and not actually used. Can we try to remove this code and just keep the gradient kernel and necessary  functions.

Reply:
I made the change and ran unit tests. Seems fine. Checking with @weixingzhang if there's any concern

Reply:
He said looks good.